cupy/statistics/histogram.py

```diff
@@ -34,7 +34,10 @@ def histogram(x, bins=10):
         if min_value == max_value:
             min_value -= 0.5
             max_value += 0.5
-        bins = cupy.linspace(min_value, max_value, bins + 1)
+        bin_type = cupy.result_type(min_value, max_value, x)
+        if cupy.issubdtype(bin_type, cupy.integer):
+            bin_type = cupy.result_Type(bin_type, float)
```

Comment:
Thanks for the catch. I noticed that this `if` block does not make sense as `bin_type` will never become integer (because `min_value` and `max_value` are always float in this context). So, removed.